### DISCUSSION TITLE
No screenshots attachments in Swift Testing tests

### DISCUSSION THREAD

**Discussion by Arideno** [pointfreeco/swift-snapshot-testing#909] [General]
**Describe the bug**
After migrating to Swift Testing, cannot see reference and failed attachments anymore.

**To Reproduce**
Call `assertSnapshot` from Swift Testing test

**Expected behavior**
Reference and failed attachments are visible.

**Environment**
  - swift-snapshot-testing version 1.17.5
  - Xcode 16.0
  - Swift 5.10
  - OS: iOS 18

**Comment by mbrandonw** [comment]
Hi @Arideno, does Swift Testing support the concept of attachments? I believe that is only an XCTest feature. And if that's the case, then this sounds like a feature request for Swift Testing and not something we can really handle.

**Comment by mbrandonw** [comment]
Hi @Arideno, I don't believe this is an issue with the library, so I am going to convert it to a discussion. Please feel free to continue the conversation over there!

**Comment by Arideno** [comment]
Hi @mbrandonw, yeah, I understand the problem. I've created a feature request in swift-testing [repo](https://github.com/swiftlang/swift-testing/issues/714)

**Comment by Arideno** [comment]
Hello, it seems they added attaches to Swift Testing [here](https://github.com/swiftlang/swift-testing/pull/770).

**Reply by mbrandonw** [reply]
Hi @Arideno, thanks for sharing! Unfortunately it's still hidden behind `@_spi(Experimental)` and so it isn't something we can take advantage of anytime soon.

**Comment by theblixguy** [comment]
Looks like this is officially implemented in Xcode 26/Swift 6.2 - https://github.com/swiftlang/swift-evolution/blob/main/proposals/testing/0009-attachments.md

Hopefully this library will adopt it soon!

**Comment by Arideno** [comment]
Hello, @mbrandonw. Is this feature planned for new Xcode 26?

**Comment by ajkolean** [comment]
I’ve opened a PR to add Swift Testing attachment support: https://github.com/pointfreeco/swift-snapshot-testing/pull/1021
Would love any feedback or testing from anyone already on Xcode 26!

**Comment by adammcarter** [comment]
Hey,

Does anyone have updates on this feature?

Thanks!

**Comment by knutanderss** [comment]
Any update on this feature? 😁

**Comment by Arideno** [comment]
Hello, everyone. This feature has been implemented in [this pr](https://github.com/pointfreeco/swift-snapshot-testing/pull/1064), so I am closing this discussion
